test/unit/utils_test.js

```diff
@@ -65,6 +65,37 @@ describe('Utils', function() {
       expect(Buffer.from(operation.value.toString(), 'base64').length).to.eql(48);
     });
 
+    it('returns challenge which follows SEP0010 ==2.0 spec', function() {
+      let keypair = StellarSdk.Keypair.random();
+
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        keypair,
+        "GBDIT5GUJ7R5BXO3GJHFXJ6AZ5UQK6MNOIDMPQUSMXLIHTUNR2Q5CFNF",
+        null,
```

Comment:
The one downside to this approach is that I can't make `anchorName` optional, since `networkPassphrase` has no default. This may mean that the best approach is having two different functions, one for each SEP-10 protocol major version.